openshift/console
# Add Pipeline Idle Icon & Add OnClick Start Pipeline Modal

**Fixes**: 
https://issues.redhat.com/browse/ODC-5047

**Analysis / Root cause**: 
Old icon was used while UX figured it out & lack of action on the Pipeline Not Started "build" decorator.

**Solution Description**: 
When implemented originally in 4.7, we didn't have the icon figured out. So we pushed it to this Story. Adds a temp icon manually while we work to get PatternFly icon integrated. Also adds a on-click functionality to the Pipeline Not Started "build" decorator, which now opens the Pipeline Start Modal.

**Screen shots / Gifs for design review**: 
![image](https://user-images.githubusercontent.com/8126518/99477288-3207db00-2920-11eb-882c-a2f8bd323fa6.png)

**Unit test coverage report**: 

New Tests:
```
 PASS  packages/dev-console/src/components/topology/components/nodes/build-decorators/__tests__/BuildDecorator.spec.tsx
  BuildDecorator renders
    ✓ expect BuildDecorator by default
    ✓ expect PipelineRunDecorator if a connectedPipeline is provided

 PASS  packages/dev-console/src/components/topology/components/nodes/build-decorators/__tests__/PipelineRunDecorator.spec.tsx 
  PipelineRunDecorator renders
    ✓ expect a log link when it contains at least one PipelineRun
    ✓ expect not to find a Link component when there is no PipelineRuns associated
    ✓ expect not to find an onClick functionality when there is a lack of permissions
```

Coverage:
```
-----------------------------------------------------------|----------|----------|----------|----------|-------------------|
File                                                       |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
-----------------------------------------------------------|----------|----------|----------|----------|-------------------|
 src/components/topology/components/nodes/build-decorators |    74.67 |    24.32 |    33.33 |    76.71 |                   |
  BuildDecorator.tsx                                       |      100 |      100 |      100 |      100 |                   |
  PipelineRunDecorator.tsx                                 |    96.77 |      100 |       50 |    96.77 |                38 |
-----------------------------------------------------------|----------|----------|----------|----------|-------------------|
```

**Test setup:**
- Add Pipeline Operator
- Add a Deployment without "Add Pipeline" checked
- Add a manual Pipeline (create using Pipeline Builder if needed)
- Add the label `app.kubernetes.io/instance=<deployment-app-name>` to the Pipeline
- See the Not Started icon in Topology

**Browser conformance**:
